### PR TITLE
SAGE-1539: add VSN filtering to API url

### PR DESCRIPTION
**Pull request by SammiSJH**
- filter VSN in url
- adjust tags column in serializers.py

### CODE REVIEW

File: api/ManifestApp/views.py [hunk 1]
```diff
@@ -10,4 +10,10 @@ def index(request):
 class NodeList(ListCreateAPIView):
 
     queryset = NodeData.objects.all().order_by("VSN")
```
Comment (jswantek) [inline api/ManifestApp/views.py:12]: do you still need this?  now that you have the `get_queryset` ?

File: api/ManifestApp/urls.py [hunk 2]
```diff
@@ -10,5 +10,5 @@
     path('',views.index, name="homepage"),
     path('api/', include(router.urls)),
     path('api-auth/', include('rest_framework.urls', namespace='rest_framework')),
-    path('api/all-nodes/', views.NodeList.as_view(), name="all-nodes"),
+    path('api/nodes/', views.NodeList.as_view(), name="nodes"),
```
Comment (jswantek) [inline api/ManifestApp/urls.py:13]: I am confused.. i expected to see

```
path('api/nodes/', views.NodeList.as_view(), name="nodes"),
path('api/nodes/<str:vsn>', views.NodeList.as_view(), name="nodes"),
```

where the first one would return all the nodes and the 2nd one would return only the 1 node specified by `<str:vsn>`

Can you please show an example of the URLs that you would use to 
a) get manifest for all the nodes
b) get the manifest for a single node (by VSN)

**Comment by seanshahkarami** [comment]
Now that we're getting closer to a stable output format, I'd encourage you to add some unit tests around this. This will help self-document the endpoints and provide usage examples of the API.

Roughly something like this should be a good start:

```python
# in your app's tests.py file

class TestAPI(TestCase):

    def setUp(self):
        # prepopulate a few made up Node, Hardware, Device, etc items
        NodeData.objects.create(...)
        ...
        
        Hardware.objects.create(...)
        ...

    def testList(self):
        r = self.client.get("/api/nodes/")
        self.assertEqual(r.status_code, 200)
        self.assertEqual(r.json(), [
            # expected json output
        ])

    def testDetail(self):
        r = self.client.get("/api/nodes/V123")
        self.assertEqual(r.status_code, 200)
        self.assertEqual(r.json(), {
            # expected json output
        })
```

Here's a (somewhat large) [set of examples](https://github.com/waggle-sensor/waggle-auth-app/blob/1eea15092feace1e1d8db95884562381e6e69b89/app/tests.py#L152) from the auth service.

It also doesn't have to be perfect the first time - the goal is to just start adding more and more coverage of the public interface. 😄

Update: We discussed merging this PR and adding unit tests as another PR since there's no plans to make any more functional changes. This will allow us to start working on the deployment settings in parallel.

File: api/ManifestApp/urls.py (hunk 2 above)
Comment (SammiSJH) [reply api/ManifestApp/urls.py:13]: a) api/nodes/
b) api/nodes/?VSN=W023 -> is this acceptable?

File: api/ManifestApp/views.py (hunk 1 above)
Comment (SammiSJH) [reply api/ManifestApp/views.py:12]: no, I should delete this

File: api/ManifestApp/urls.py (hunk 2 above)
Comment (jswantek) [reply api/ManifestApp/urls.py:13]: I would really like it to be

`api/nodes/`
`api/nodes/<VSN>`  (ex. `api/nodes/W023`)

**Comment by iperezx** [comment]
I made this issue for the unit testing: https://github.com/waggle-sensor/beekeeper-manifest/issues/4

**Pull request merged**